### ISSUE TITLE
StableDiffusionInpaintPipeline missing from_single_file

### ISSUE THREAD

**Issue by ballenvironment** [huggingface/diffusers#4002]
**Is your feature request related to a problem? Please describe.**
The inpaint pipeline is missing `from_single_file` method, which is inconsistent with the other pipelines.

**Describe the solution you'd like**
```py
generator = StableDiffusionInpaintPipeline.from_single_file(model)
```

**Describe alternatives you've considered**
I'm currently doing this as an alternative, but it's ugly code.
```py
generator = StableDiffusionImg2ImgPipeline.from_single_file(model)
saved_vae = generator.vae
saved_text_encoder = generator.text_encoder
saved_tokenizer = generator.tokenizer
saved_unet = generator.unet
saved_scheduler = generator.scheduler
saved_safety_checker = generator.safety_checker
saved_feature_extractor = generator.feature_extractor
generator = StableDiffusionInpaintPipeline(vae=saved_vae, text_encoder=saved_text_encoder, tokenizer=saved_tokenizer, 
            unet=saved_unet, scheduler=saved_scheduler, safety_checker=saved_safety_checker, feature_extractor=saved_feature_extractor)
```

**Additional context**

**Comment by sayakpaul** [comment]
If we check here:

https://github.com/huggingface/diffusers/blob/6632823690707604ef364e4d2fa02269bf1a7d07/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_img2img.py#L102C70-L102C89

the `StableDiffusionImg2ImgPipeline` is being subclassed from `FromSingleFileMixin` as well which is not the case for `StableDiffusionInpaintPipeline`:

https://github.com/huggingface/diffusers/blob/6632823690707604ef364e4d2fa02269bf1a7d07/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint.py#L156C7-L156C37

Would you maybe like to open a PR to help us improve this?

**Comment by patrickvonplaten** [comment]
Fixed in #4041

**Comment by ballenvironment** [comment]
Much appreciated!
